### PR TITLE
[Part 1] Policy updates - Remove RetryStrategy from Retry Policy 

### PR DESCRIPTION
**Pull request by samvaity**
This PR removes the dependency on `RetryStrategy`.
Make RetryPolicy configurable only via RetryOptions.

Code changes:
- Removed `RetryPolicy.RetryStrategy`
- Removed implementation models `ExponentialBackoffDelay` and `FixedDelay`

### CODE REVIEW

File: sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/policy/RetryPolicy.java [hunk 1]
```diff
@@ -63,11 +68,11 @@ public class RetryPolicy implements HttpPipelinePolicy {
     }
 
     /**
-     * Creates {@link RetryPolicy} using {@link ExponentialBackoffDelay} as the {@link RetryStrategy} and defaulting to
-     * three retries.
+     * Creates {@link RetryPolicy} using ExponentialBackoffDelay strategy, defaulting to
```
Comment (srnagar) [inline sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/policy/RetryPolicy.java:71]:
```suggestion
     * Creates {@link RetryPolicy} using exponential backoff delay, defaulting to
```

File: sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/policy/RetryPolicy.java [hunk 2]
```diff
@@ -77,25 +82,30 @@ public RetryPolicy() {
      * @throws NullPointerException If {@code retryOptions} is null.
      */
     public RetryPolicy(RetryOptions retryOptions) {
-        this(ImplUtils.getRetryStrategyFromOptions(retryOptions), retryOptions.getMaxRetries(),
-            retryOptions.getDelayFromHeaders());
+        this(retryOptions.getBaseDelay(), retryOptions.getMaxDelay(), retryOptions.getFixedDelay(),
```
Comment (srnagar) [inline sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/policy/RetryPolicy.java:85]: If retryOptions.getBaseDelay is `null` and getFixedDelay is also `null`, we should default to `DEFAULT_BASE_DELAY` and same for other values.

File: sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/implementation/util/ImplUtils.java [hunk 3]
```diff
@@ -180,120 +162,6 @@ public static byte[] byteBufferToArray(ByteBuffer byteBuffer) {
...
     }
 
-    /**
-     * Utility method for parsing a {@link URL} into a {@link UrlBuilder}.
-     *
-     * @param url The URL being parsed.
-     * @param includeQuery Whether the query string should be excluded.
-     * @return The UrlBuilder that represents the parsed URL.
-     */
-    public static UrlBuilder parseUrl(URL url, boolean includeQuery) {
```
Comment (srnagar) [inline sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/implementation/util/ImplUtils.java:190]: Are these two utility methods no longer needed?

Comment (samvaity) [reply sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/implementation/util/ImplUtils.java:190]: yeah, they were unused in the current state of repo. We can always add back if needed.

File: sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/policy/RetryPolicy.java [hunk 4]
```diff
@@ -99,10 +102,19 @@ public RetryPolicy(RetryOptions retryOptions) {
...
      */
     RetryPolicy(Duration baseDelay, Duration maxDelay, Duration fixedDelay, int maxRetries,
-        Function<Headers, Duration> delayFromHeaders, Predicate<RequestRetryCondition> shouldRetryCondition) {
+                Function<Headers, Duration> delayFromHeaders, Predicate<RequestRetryCondition> shouldRetryCondition) {
+        if (fixedDelay == null && baseDelay == null) {
+            this.baseDelay = DEFAULT_BASE_DELAY;
+            if (maxDelay == null) {
+                this.maxDelay = DEFAULT_MAX_DELAY;
+            } else {
+                this.maxDelay = maxDelay;
```
Comment (srnagar) [inline sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/policy/RetryPolicy.java:111]: Do we allow only setting maxDelay without setting baseDelay?

**Pull request closed without merging**